From 4b3276485bb5fc46166f0ede5e026e81760d2e23 Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Fri, 14 Sep 2012 14:07:40 +0800
Subject: [PATCH 217/304] ENGR00224187 imx6sl keypad ENTER does not work

1. add idc file for imx-keypad, then keypad layout will be loaded.
2. move the imx-keypad.idc, imx-keypad.kl to common/input

Signed-off-by: b02247 <b02247@freescale.com>
---
 common/input/imx-keypad.idc | 25 +++++++++++++++++++++++++
 common/input/imx-keypad.kl  | 11 +++++++++++
 imx6/evk_6sl.mk             |  3 ++-
 3 files changed, 38 insertions(+), 1 deletion(-)
 create mode 100644 common/input/imx-keypad.idc
 create mode 100644 common/input/imx-keypad.kl

diff --git a/common/input/imx-keypad.idc b/common/input/imx-keypad.idc
new file mode 100644
index 0000000..2ccc0a2
--- /dev/null
+++ b/common/input/imx-keypad.idc
@@ -0,0 +1,25 @@
+# Copyright (C) 2010 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+touch.deviceType = touchScreen
+touch.orientationAware = 1
+
+keyboard.layout = imx-keypad
+keyboard.characterMap = Generic
+keyboard.orientationAware = 1
+keyboard.builtIn = 1
+
+cursor.mode = navigation
+cursor.orientationAware = 1
diff --git a/common/input/imx-keypad.kl b/common/input/imx-keypad.kl
new file mode 100644
index 0000000..b2e3ec9
--- /dev/null
+++ b/common/input/imx-keypad.kl
@@ -0,0 +1,11 @@
+key 158    BACK
+key 116    POWER           WAKE
+key 139    MENU
+key 412    BACK
+key 102    HOME
+key 353    DPAD_CENTER
+key 103    DPAD_UP
+key 105    DPAD_LEFT
+key 106    DPAD_RIGHT
+key 108    DPAD_DOWN
+
diff --git a/imx6/evk_6sl.mk b/imx6/evk_6sl.mk
index f7be995..a50c8bb 100644
--- a/imx6/evk_6sl.mk
+++ b/imx6/evk_6sl.mk
@@ -12,7 +12,8 @@ PRODUCT_COPY_FILES += \
 	device/fsl/evk_6sl/required_hardware.xml:system/etc/permissions/required_hardware.xml \
 	device/fsl/evk_6sl/vold.fstab:system/etc/vold.fstab \
 	device/fsl/evk_6sl/init.rc:root/init.freescale.rc \
-	device/fsl/evk_6sl/mxckpd.kl:system/usr/keylayout/mxckpd.kl \
+	device/fsl/common/input/imx-keypad.idc:system/usr/idc/imx-keypad.idc \
+	device/fsl/common/input/imx-keypad.kl:system/usr/keylayout/imx-keypad.kl \
 	device/fsl/evk_6sl/audio_policy.conf:system/etc/audio_policy.conf \
 	device/fsl/evk_6sl/audio_effects.conf:system/vendor/etc/audio_effects.conf
 
-- 
1.8.0

